### PR TITLE
Query methods issue with no criteria bug fix for Spring JPA Keywords

### PR DESCRIPTION
**Pull request by trande4884**
# Description

Fixes bug #40092 

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java [hunk 1]
```diff
@@ -226,7 +226,10 @@ private String generateQueryBody(@NonNull Criteria criteria, @NonNull List<Pair<
     @NonNull
     private Pair<String, List<Pair<String, Object>>> generateQueryBody(@NonNull CosmosQuery query, @NonNull final AtomicInteger counter) {
         final List<Pair<String, Object>> parameters = new ArrayList<>();
-        String queryString = this.generateQueryBody(query.getCriteria(), parameters, counter);
+        String queryString = "";
+        if (query.getCriteria() != null) {
+            queryString = this.generateQueryBody(query.getCriteria(), parameters, counter);
```
Comment (kushagraThapar) [inline sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java:231]: So if the criteria is null, will the query body be completely empty? how will that work for non-criteria queries?

Comment (FabianMeiswinkel) [reply sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java:231]: I was confused initially as well - but looks like it is just coming from unintuitive variable names - generateQueryBody does not generate the query text/body but just the where clause

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks!

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java (hunk 1 above)
Comment (trande4884) [reply sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java:231]: Correct I was confused at first also, many of the new tests I added do have criteria as null

Comment (kushagraThapar) [reply sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java:231]: Awesome, thanks @trande4884

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**